identity/src/main/java/com/android/identity/util/SimpleApplicationData.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.util;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.android.identity.internal.Util;
+
+import java.io.ByteArrayInputStream;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import co.nstant.in.cbor.CborBuilder;
+import co.nstant.in.cbor.CborDecoder;
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.builder.MapBuilder;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.UnicodeString;
+
+/**
+ * An implementation of {@link ApplicationData} using a LinkedHashMap to back the key-value pairs.
+ */
+public class SimpleApplicationData implements ApplicationData {
+    private Map<String, byte[]> mApplicationData = new LinkedHashMap<>();
+    private ChangeListener mListener;
+    private Executor mListenerExecutor;
+
+    /**
+     * Creates a new SimpleApplicationData.
+     */
+    public SimpleApplicationData() {
+
+    }
+
+    /**
+     * Set the listener to be used for notification when changes are made to the
+     * {@link SimpleApplicationData}.
+     *
+     * <p>This may be called multiple times but only one listener is active at one time.
+     *
+     * @param listener the listener or <code>null</code> to stop listening.
+     * @param executor a {@link Executor} to do the call in or <code>null</code> if
+     *                 <code>listener</code> is <code>null</code>.
+     * @throws IllegalStateException if {@link Executor} is {@code null} for a non-{@code null}
+     * listener.
+     */
+    public void setChangeListener(@Nullable ChangeListener listener, @Nullable Executor executor) {
+        if (listener != null && executor == null) {
+            throw new IllegalStateException("Passing null Executor for non-null Listener");
+        }
+        mListener = listener;
+        mListenerExecutor = executor;
+    }
```

Comment:
I don't think we have any users of this class that wouldn't call `setChangeListener()` so we should just take the listener and executor in the constructor.

Reply:
The `decodeFromCbor` method creates a new `SimpleApplicationData`, so in order to move the listener and executor to the constructor, `decodeFromCbor` would also need to take in the listener and executor. IMO, it's a bit weird to assume all users of `decodeFromCbor` would also have a listener/executor in mind but interested to hear your thoughts